pkg/controller/bucket/bucket_controller.go

```diff
@@ -99,18 +100,32 @@ func (bl *bucketListener) Add(ctx context.Context, obj *v1alpha1.Bucket) error {
 	}
 
 	req := osspec.ProvisionerCreateBucketRequest{
-		BucketName: obj.Spec.BucketRequest.Name,
+		BucketContext: map[string]string{},
+	}
+	if len(obj.Spec.Parameters["BucketInstanceName"]) > 0 {
+		req.BucketName = obj.Spec.Parameters["BucketInstanceName"]
+	} else if len(obj.Spec.Parameters["BucketPrefix"]) > 0 {
+		req.BucketContext["BucketPrefix"] = obj.Spec.Parameters["BucketPrefix"]
+	}
+
+	anonAccessMode := obj.Spec.Parameters["AnonymousAccessMode"]
+	if len(anonAccessMode) > 0 {
+		req.AnonymousBucketAccessMode = osspec.ProvisionerCreateBucketRequest_AnonymousBucketAccessMode(osspec.ProvisionerCreateBucketRequest_AnonymousBucketAccessMode_value[anonAccessMode])
 	}
 
 	switch obj.Spec.Protocol.Name {
 	case v1alpha1.ProtocolNameS3:
 		req.Region = obj.Spec.Protocol.S3.Region
```

Comment:
Protocol in the Spec is RequestedProtocol, may not have the fields you are expecting.

Reply:
In the bucket it is supposed to be the full protocol spec, not just RequestedProtocol.  From the api definition:
type BucketSpec struct {
...
        Protocol          Protocol `json:"protocol"`
...